### PR TITLE
8345323: Parallel GC does not handle UseLargePages and UseNUMA gracefully

### PR DESCRIPTION
**Pull request by albertnetymk**
This patch reverts the default value of `OldSize` to its previous setting prior to being obsoleted in [JDK-8333962](https://bugs.openjdk.org/browse/JDK-8333962). The change addresses an issue where `OldSize` being set to zero results in a default `MinHeapSize` that is too small to handle LargePages correctly. This problem is exemplified by `ParallelArguments::initialize_heap_flags_and_sizes`, as identified in [JDK-8345323](https://bugs.openjdk.org/browse/JDK-8345323).

Changing the default value of `OldSize` may have broader implications due to the complexity of the logic that determines default values for various flags. Altering one default can lead to cascading effects and potential breakages elsewhere. For these reasons, this patch restores the previous default value of `OldSize` to mitigate such risks.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345323](https://bugs.openjdk.org/browse/JDK-8345323): Parallel GC does not handle UseLargePages and UseNUMA gracefully (**Bug** - P3)(⚠️ The fixVersion in this issue is [24] but the fixVersion in .jcheck/conf is 25, a new backport will be created when this pr is integrated.)

### Reviewers
 * [Stefan Johansson](https://openjdk.org/census#sjohanss) (@kstefanj - **Reviewer**)
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22575/head:pull/22575` \
`$ git checkout pull/22575`

Update a local copy of the PR: \
`$ git checkout pull/22575` \
`$ git pull https://git.openjdk.org/jdk.git pull/22575/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22575`

View PR using the GUI difftool: \
`$ git pr show -t 22575`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22575.diff">https://git.openjdk.org/jdk/pull/22575.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22575#issuecomment-2520143611)
</details>

### CODE REVIEW

File: src/hotspot/share/gc/shared/genArguments.cpp [hunk 1]
```diff
@@ -37,7 +37,11 @@ size_t MinNewSize = 0;
 size_t MinOldSize = 0;
 size_t MaxOldSize = 0;
 
-size_t OldSize = 0;
+// If InitialHeapSize or MinHeapSize is not set on cmdline, this variable,
+// together with NewSize, are used to derive them.
```
Comment (kstefanj) [inline src/hotspot/share/gc/shared/genArguments.cpp:41]:
```suggestion
// together with NewSize, is used to derive them.
```

**Review by kstefanj — APPROVED** [review]
Looks good, thanks for doing the fix.

Just a small comment on the comment.

File: src/hotspot/share/gc/shared/genArguments.cpp [hunk 2]
```diff
@@ -37,7 +37,11 @@ size_t MinNewSize = 0;
 size_t MinOldSize = 0;
 size_t MaxOldSize = 0;
 
-size_t OldSize = 0;
+// If InitialHeapSize or MinHeapSize is not set on cmdline, this variable,
+// together with NewSize, is used to derive them.
+// Using the same value when it was a configurable flag to avoid breakage.
+// See more in JDK-8345323
```
Comment (tschatzl) [inline src/hotspot/share/gc/shared/genArguments.cpp:43]: I do not like referrals to the bug tracker in the code, and/or referring to some code the past ("when it was configurable").
Better explicitly state the problem with heap sizing and large pages and file a follow-up RFE (not mentioning it here).

Comment (tschatzl) [reply src/hotspot/share/gc/shared/genArguments.cpp:43]: I.e. something like "If the default value of OldSize is too small, then ..., leading to the generations not aligned and not being able to allocate large pages" or so.

**Review by tschatzl — APPROVED** [review]
lgtm

**Comment by albertnetymk** [comment]
Thanks for review.

/integrate

**Pull request closed without merging**